kit/Kit.cpp

```diff
@@ -1152,6 +1152,17 @@ class Document final : public DocumentManagerInterface
             }
             return;
         }
+        else if (type == LOK_CALLBACK_DOCUMENT_PASSWORD_RESET) {
+            Document* document = dynamic_cast<Document*>(descriptor->getDoc());
+            if (document)
+            {
+                document->_isDocPasswordProtected = p ? !payload.empty() : false;
+                document->_haveDocPassword = document->_isDocPasswordProtected;
+                document->_docPassword = payload;
+                document->_docPasswordType = PasswordType::ToView;
```

Comment:
Does this imply that we can't reset the password to modify a document?

(Perhaps another reason why reusing `LOK_CALLBACK_DOCUMENT_PASSWORD` and `LOK_CALLBACK_DOCUMENT_PASSWORD_TO_MODIFY` is a good idea.)